### DISCUSSION TITLE
servo pulsing issue

### DISCUSSION THREAD

**Discussion by atomGit** [EdgeTX/edgetx#5361] [Q&A] [answered]
hi all - i'm on ETX 2.10.2 and i have a problem with servo pulsing when the Tx stick is below 80% or so

i'm driving 4 [GoBuilda 2000 series servos](https://www.servocity.com/2000-series-dual-mode-servo-25-2/) in continuous rotation mode and they all pulse, meaning that they change their speed, slightly, about 20-30 times for each rotation

i have an old HiTech servo programmer and when i drive one of the servos with that, it works fine, though it's 4.8v whereas i'm normally driving them with 7.4v - i don't know if that makes any difference

the Tx and Rx are a RadioMaster Boxer, ELRS, both with the latest firmware, and i think i've tried messing with every setting that i thought might cause the problem, to no avail

i'm stumped

**Comment by pfeerick** [comment]
I don't know anything about those servos, but I would probably start
looking at the ELRS servo output options on the webUI, as well as checking
the signal requirements for those servos ... ELRS doesn't play nicely with
some servo applications.

**Comment by atomGit** [comment]
i tried messing with the all frequency settings from 50Hz to DShot - no dice

the radio is ELRS, so the protocol needs to be CRSF i assume, so i didn't mess with that

the PWM range for the servo according to the spec is ...

```
Max PWM Range 	500-2500μsec
Max PWM Range (Continuous) 	900-2100µsec
```

...so it changes for continuous mode, but i'm not seeing were this can be changed on the Tx or Rx

**Comment by pfeerick** [comment]
ELRS is limited to PWM 885us to 2115us - there is nothing we can do to change that on the TX. 

To go beyond that, it looks like you also will need https://github.com/ExpressLRS/ExpressLRS/pull/2773, or a "servo stretcher" like https://www.servocity.com/180-servo-stretcher/ - there are a few different types if you search for that name.

**Comment by atomGit** [comment]
> ELRS is limited to PWM 885us to 2115us ...

that's pretty much in line with what the servo expects in continuous mode

i contacted RadioMaster also ... maybe they'll have something to say - maybe i'll need to buy an external 900MHz module and Rx and try that

**Reply by pfeerick** [reply]
Have you enabled extended limits (toggle at the top of the outputs page) and lowered/increased the min/max range for that channels output? As by default you will be limited to 988-2012 on the handset, but should be able wind it out to -120/120 to get the ELRS max values...

**Comment by atomGit** [comment]
i enabled the extended limits and set the ranges to -120/+120 (they will go to 150)

~~the servos run faster of course, but~~ the problem remains

**Comment by atomGit** [comment]
tech at RadioMaster got back to me...

> Traditional model remote control PWM value 1050-1950us, maximum 1000-2000us.
While EdgeTX output PWM value 988-2021us, maximum 732-2268us. This may be beyond the limit of recognition of the steering gear.
>
> You can modify the Min and Max values in EdgeTX's MDL-OUTPUTS page, narrowing down the output channel values appropriately, such as Min: -88, Max: +88, to match the conventional signal 1050-1950us.

i dynamically adjusted the channel output range while driving the servo and while this of course speeds up/slows down servo rotation, it has no effect on the pulsing

right now the biggest question is whether this is a radio issue, and ELRS issue, or a servo-radio-ELRS compatibility issue

neither ServoCity nor RadioMaster has any answers yet and it seems this isn't a PWM usec range issue - at ~80% stick or less, the servos are acting as if they are being fed 2 alternating usec values that switch ~20-30 times every rotation

**Comment by atomGit** [comment]
if anyone can reproduce this, i'd sure appreciate it 

to briefly recap, the issue is that the GoBuilda 2000 Series Dual Mode 
Servo doesn't run smoothly when it's operated in continuous mode - it 
changes speed slightly about 30 times per rotation - this occurs from 
roughly 1-80% stick input, then smooths out and runs fine as the stick 
approaches 100%

hardware needed...

* Boxer ELRS radio - firmware version doesn't seem to matter
* ER8 2.4GHz ELRS PWM Receiver
* 7.4v battery
* 2000 Series Dual Mode Servo (25-2, Torque) 
(https://www.servocity.com/2000-series-dual-mode-servo-25-2/)
* 3102 Series Dual Mode Servo Programmer (1-1) 
(https://www.servocity.com/3102-series-dual-mode-servo-programmer-1-1/)

steps to reproduce...

1. using the programmer, set the servo in continuous mode
2. connect servo to Rx channel 1-4
3. set up Tx/Rx for ELRS and power everything up
4. set the radio so that the servo is operated by rudder or aileron 
stick, although it may not matter
5. feed some stick input to the servo
6. observe that the servo doesn't run smoothly at small stick inputs

**Comment by inventor7777** [comment]
Have you tried a different receiver?

**Comment by atomGit** [comment]
i have only one

**Reply by inventor7777** [reply]
Have you tried resetting the receiver and rebinding it?

**Comment by atomGit** [comment]
been meaning to try that - did it now and no change

**Comment by inventor7777** [comment]
It _sounds_ like a receiver problem to me. But I’m no ELRS expert so…yeah.

**Comment by atomGit** [comment]
from the discussion here i suspect it may be an issue with these particular servos, but i'm unsure

apparently there's a way to do a rudimentary test of the PWM signal being sent from the receiver using a multimeter set to Hz and testing the signal and ground pins on the receiver but that didn't yield any results

still trying to decide what my next step will be, perhaps introduce a flight controller into the mix and drive the servos from that ... i'm waiting to hear more from ServoCity first

**Reply by pfeerick** [reply]
Either/or... I would be suspect the receiver or the servos, or simply that particular combination... the ER8 *could* be doing something a bit "weird" with the PWM output that that particular servo doesn't like... whereas a Matek [CSRF to PWM](https://www.mateksys.com/?portfolio=crsf-pwm) style receiver/converter might be just fine as it runs the PWM itself, ratherthan the receiver having to do receiver stuff AND drive the PWM (although I don't recommend those Matek RX at all, as it brings some other unnecessary complexity and annoyances, as well having made/sold some very bad RXs in the past).

**Comment by atomGit** [comment]
quoting myself here from over at the [ELRS repo](https://github.com/ExpressLRS/ExpressLRS/discussions/2876)...

> i have an old HiTech servo programmer and when i drive one of the servos with that, it works fine, though it's 4.8v whereas i'm normally driving them with 7.4v - i don't know if that makes any difference

**it absolutely does** - feeding the ELRS receiver with 5.1v, the servos run perfectly - i can't test anything between 5.1 and 7.4 at the moment

**Comment by atomGit** [comment, marked as answer]
FINALLY got a definitive answer (from Ethan this time <-- thanks Ethan :1st_place_medal: ) at ServoCity - the newest batch of the [GoBuilda 2000 series servos](https://www.servocity.com/2000-series-dual-mode-servo-25-2/) (and possibly others in that product line ???) do not operate correctly when they are run in continuous mode and the voltage is 7.4, or thereabouts - they do run fine at 5v and 6v though

**Reply by pfeerick** [reply]
Ouch! Glad you got to the bottom of it! :)

**Reply by inventor7777** [reply]
Hopefully they can send you some new servos!

**Comment by atomGit** [comment]
they're sending new servos (new = older batch)
